LFP_analysis/Laser_LFP_Parse_Final.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jul  6 10:06:32 2017
+
+@author: Bradly
+"""
+#Import necessary tools
+import numpy as np
+import easygui
+import tables
+import os
+import matplotlib.pyplot as plt
+from scipy import signal
+from scipy.signal import periodogram
+
+#Get name of directory where the data files and hdf5 file sits, and change to that directory for processing
+dir_name = easygui.diropenbox()
+os.chdir(dir_name)
+
+#Look for the hdf5 file in the directory
+file_list = os.listdir('./')
+hdf5_name = ''
+for files in file_list:
+	if files[-2:] == 'h5':
+		hdf5_name = files
+
+#Open the hdf5 file and create list of child paths
+hf5 = tables.open_file(hdf5_name, 'r+')
+lfp_nodes = hf5.list_nodes('/Parsed_LFP')
+laser_nodes = hf5.list_nodes('/spike_trains')
+
+# Create a LFP_laser group in the hdf5 file
+try:
+    hf5.remove_node('/LFP_Lasers', recursive = True)
+except:
+    pass
+hf5.create_group('/', 'LFP_Lasers')
+
+#Establish number of tastes and laser_combination variables
+taste_count = len(laser_nodes)
+laser_comb_count = len(laser_nodes)
+trials = hf5.root.ancillary_analysis.trials[1,:]
+trials_per_taste_las_cond = int(len(trials)/taste_count)
+
+def blockshaped(arr, nrows, ncols):
+    """
+    Return an array of shape (n, nrows, ncols) where
+    n * nrows * ncols = arr.size
+    
+    If arr is a 2D array, the returned array should look like n subblocks with
+    each subblock preserving the "physical" layout of arr.
+    """
+    h, w = arr.shape
+    return (arr.reshape(h//nrows, nrows, -1, ncols)
+               .swapaxes(1,2)
+               .reshape(-1, nrows, ncols))
+
+# Create a taste group in the hdf5 file
+for i in range(taste_count):
+    taste = 'taste_' + str(i)         
+    hf5.create_group('/LFP_Lasers', str.split(taste, '/')[-1])    
+    
+#Create loop based on number of laser conditions and create groupings of blank arrays to fill
+for z in range(taste_count):        #Where z indicates dig_in (laser condition)

```

Comment:
I'm very confused about what's happening in the code from here on. I feel that this is far more complicated than it needs to be :'( I would basically suggest something along these lines:
```
# Get the laser conditions
laser_conditions = hf5.root.ancillary_analysis.laser_combination_d_l[:]
laser_trials = hf5.root.ancillary_analysis.trials[:]

# Get number of tastes
num_tastes = len(hf5.list_nodes('/spike_trains'))
# Get number of trials for each taste (assume equal number of trials for each)
num_trials = hf5.root.spike_trains.dig_in_0.spike_array[:].shape[....fill in the right numbers here]

# Run through the tastes and laser conditions, and fill up an array with the data
for i in num_tastes:
   for j in laser_conditions.shape[0]:
      # Pick the right trial numbers here, and fill them in your array
```
Basically, this part needs drastic modifications!